File: src/ex2/src/my_subscriber.py
#!/usr/bin/python3
import rospy 
from std_msgs.msg import Int32, String


def text_callback(msg):
    text = msg.data
    rospy.loginfo("Received text: " + text)


def main():
    rospy.init_node('my_subscriber')
    sub = rospy.Subscriber('text', String, text_callback)
    rospy.spin()


if __name__ == '__main__': 
  try: 
      main()
  except rospy.ROSInterruptException: 
      pass

File: src/ex3/src/my_subscriber.cpp
#include <iostream>
#include <ros/ros.h>
#include <std_msgs/Float32.h>

using namespace std;

float sum = 0;

void sinus_cb(const std_msgs::Float32::ConstPtr& msg) {  
	
	sum += msg->data;
	ROS_INFO("Received: %f Sum: %f", msg->data, sum);
}
	
int main(int argc, char* argv[]) {  

	ros::init(argc, argv, "my_subscriber");  
	ros::NodeHandle nh;  
	ros::Subscriber sub = nh.subscribe<std_msgs::Float32>("sinus", 10, sinus_cb);  
	ros::spin();  
	
	return 0;
}
